x-pack/legacy/plugins/ml/server/routes/anomaly_detectors.ts

```diff
@@ -0,0 +1,460 @@
+      }
+    })
+  );
+
+  /**
+   * @apiGroup AnomalyDetectors
+   *
+   * @api {get} /api/ml/anomaly_detectors/_stats Get anomaly detection stats
+   * @apiName GetAnomalyDetectorsStats
+   * @apiDescription Returns anomaly detection jobs statistics.
+   *
+   * @apiSuccess {Number} count
+   * @apiSuccess {Object[]} jobs
+   */
+  router.get(
+    {
+      path: '/api/ml/anomaly_detectors/_stats',
+      validate: false,
+    },
+    licensePreRoutingFactory(xpackMainPlugin, async (context, request, response) => {
+      try {
+        const results = await context.ml!.mlClient.callAsCurrentUser('ml.jobStats');
+        return response.ok({
+          body: { ...results },
+        });
+      } catch (e) {
+        return response.customError(wrapError(e));
+      }
+    })
+  );
+
+  /**
+   * @apiGroup AnomalyDetectors
+   *
+   * @api {get} /api/ml/anomaly_detectors/:jobId/_stats Get stats for requested anomaly detection job
+   * @apiName GetAnomalyDetectorsStatsById
+   * @apiDescription Returns anomaly detection job statistics.
+   *
+   * @apiParam {String} jobId Job ID.
+   */
+  router.get(
+    {
+      path: '/api/ml/anomaly_detectors/{jobId}/_stats',
+      validate: {
+        params: schema.object({
+          jobId: schema.string(),
+        }),
+      },
+    },
+    licensePreRoutingFactory(xpackMainPlugin, async (context, request, response) => {
+      try {
+        const { jobId } = request.params;
+        const results = await context.ml!.mlClient.callAsCurrentUser('ml.jobStats', { jobId });
+        return response.ok({
+          body: { ...results },
+        });
+      } catch (e) {
+        return response.customError(wrapError(e));
+      }
+    })
+  );
+
+  /**
+   * @apiGroup AnomalyDetectors
+   *
+   * @api {put} /api/ml/anomaly_detectors/:jobId Instantiate an anomaly detection job
+   * @apiName CreateAnomalyDetectors
+   * @apiDescription This API creates an anomaly detection job that performs an analysis
+   *                 on the source index and stores the outcome in a destination index.
+   *
+   * @apiParam {String} jobId Job ID.
+   */
+  router.put(
+    {
+      path: '/api/ml/anomaly_detectors/{jobId}',
+      validate: {
+        params: schema.object({
+          jobId: schema.string(),
+        }),
+        body: schema.object({ ...anomalyDetectionJobSchema }),
+      },
+    },
+    licensePreRoutingFactory(xpackMainPlugin, async (context, request, response) => {
+      try {
+        const { jobId } = request.params;
+        const results = await context.ml!.mlClient.callAsCurrentUser('ml.addJob', {
+          jobId,
+          body: request.body,
+        });
+        return response.ok({
+          body: { ...results },
+        });
+      } catch (e) {
+        return response.customError(wrapError(e));
+      }
+    })
+  );
+
+  /**
+   * @apiGroup AnomalyDetectors
+   *
+   * @api {put} /api/ml/anomaly_detectors/:jobId/_update Update an anomaly detection job
```

Comment:
but could be a typo in the route handler, probably PUT is used for an update

Reply:
This was a `POST` in the original route file 🤔 I'd be reluctant to change it in this PR in case of unforeseen side effects. But definitely something to look into separately and see if we can update to a `PUT` as I agree, it makes more sense. 👌

Reply:
The `POST` matches the method used in the back-end Java endpoint  - https://www.elastic.co/guide/en/elasticsearch/reference/master/ml-update-job.html, so I think you are right to keep it as `POST` here.